### DISCUSSION TITLE
weasel 修改代码并build之后如何立即生效

### DISCUSSION THREAD

**Discussion by kinnplh** [rime/home#1512] [Q&A] [answered]
各位大佬！我最近在看weasel代码，同时改着代码玩。我发现在修改代码并运行`build.bat weasel` 之后：
1. 需要在`WeaselSetup.exe`中点击“修改文件夹”，之后修改会生效。
2. 但是，如果修改涉及到前端显示，则即使执行了第一步，相关修改仍然不能立即生效。在重启电脑之后，相关修改才能生效。

想请教一下如何才能够在不重启的前提下立即生效？重启电脑太耗费时间了！
感谢！！！

**Comment by fxliang** [comment, marked as answer]
如果涉及修改tsf关联的内容（WeaselIPC/WeaselTSF/WeaselUI)大致以下步骤，
WeaselSetup删除输入法，安装输入法
重启WeaselServer
重启应用，则新安装的tsf dll生效 

如果只修改 WeaselServer/RimeWithWeasel/WeaselIPCServer，则可以只重新启动算法服务即可

注意：如果修改了ipc的内容，按键输入潜在可能会引起旧的应用（已经装载Weasel的dll）崩溃，如果不确定可以重新安装后注销重新登陆一次可完全避免
